cmd/create.go

```diff
@@ -20,114 +22,177 @@ func init() {
-		fn.WithRepository(cfg.Repository),
-		fn.WithVerbose(cfg.Verbose))
+func newCreateClient(cfg createConfig) (*fn.Client, error) {
+	client := fn.New(
+		fn.WithRepositories(cfg.Repositories), // path to repositories in disk
+		fn.WithRepository(cfg.Repository),     // URI of repository override
+		fn.WithVerbose(cfg.Verbose))           // verbose logging
+	return client, nil
 }
 
-// createClientFn is a factory function which returns a Client suitable for
-// use with the Create command.
-type createClientFn func(createConfig) *fn.Client
-
 // NewCreateCmd creates a create command using the given client creator.
 func NewCreateCmd(clientFn createClientFn) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "create [PATH]",
-		Short: "Create a function project",
-		Long: `Create a function project
-
-Creates a new function project in PATH, or in the current directory if no PATH is given. 
-The name of the project is determined by the directory name the project is created in.
-`,
-		Example: `
-# Create a Node.js function project in the current directory, choosing the
-# directory name as the project's name.
-kn func create
-
-# Create a Quarkus function project in the directory "sample-service". 
-# The directory will be created in the local directory if non-existent and 
-# the project is called "sample-service"
-kn func create --runtime quarkus sample-service
-
-# Create a function project that uses a CloudEvent based function signature
-kn func create --template events sample-events
-	`,
+		Use:   "create",
+		Short: "Create a Function Project",
+		Long: `
+NAME
+	{{.Prefix}}func create - Create a Function project.
+
+SYNOPSIS
+	func create [-l|--language] [-t|--template] [-r|--repository]
+	             [-c|--confirm]  [-v|--verbose] [path]
+
+DESCRIPTION
+	Creates a new Function project.
+
+	  $ {{.Prefix}}func create -l go -t http
+
+	Creates a Function in the current directory '.' which is written in the
+	language/runtime 'Go' and handles HTTP events.
+
+	If [path] is provided, the Function is initialized at that path, creating
+	if necessary.
+
+	To complete this command interactivly, use --confirm (-c):
+	  $ {{.Prefix}}func create -c
+
+	Available Language Runtimes and Templates:
+{{ .Options | indent 2 " " | indent 1 "\t" }}
+
+	To install more language runtimes and their templates see '{{.Prefix}}func repository'.
+
+EXAMPLES
+	o Create a Node.js Function (the default language runtime) in the current
+	  directory (the default path) which handles http events (the default
+	  template).
+	  $ {{.Prefix}}func create
+
+	o Create a Node.js Function in the directory 'myfunc'.
+	  $ {{.Prefix}}func create myfunc
+
+	o Create a Go Function which handles Cloud Events in ./myfunc.
+	  $ {{.Prefix}}func create -l go -t events myfunc
+		`,
 		SuggestFor: []string{"vreate", "creaet", "craete", "new"},
-		PreRunE:    bindEnv("runtime", "template", "repository", "confirm"),
+		PreRunE:    bindEnv("language", "template", "repository", "confirm"),
```

Comment:
The flag short name and long name was inconsistent and causing confusion.  So, since `-r` was already taken by `--repository`, using `--language` for the flag, but in all user-facing text being careful to refer to it by its full name "Language Runtime".  I think this is probably good for new users as well, because they will recognize "language", but the nuance of "runtime" may not be immediately obvious.